libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFSDKDatasharingHelper.m

```diff
@@ -68,6 +67,11 @@ - (void)setAppGroupEnabled:(BOOL)appGroupEnabled {
     NSUserDefaults *sharedDefaults = [[NSUserDefaults alloc] initWithSuiteName:self.appGroupName];
     [sharedDefaults setBool:appGroupEnabled forKey:kAppGroupEnabled];
     [sharedDefaults synchronize];
+    if(appGroupEnabled) {
+        [self migrateUserDefaultsToAppContainer:sharedDefaults];
+    } else {
+        [self migrateFromAppContainerToUserDefaults:sharedDefaults];
```

Comment:
We had talked in the past about how this probably wouldn't work, as it's entirely possible that the developer would have turned off app group sharing, prior to retreating from shared settings in the app.

I think, at a minimum, if we want to try to support the "backtrack" case, we should log a warning which states that, if you've already disabled app group entitlements in your app, you will not be able to move data back into its prior, non-shared state.